Lib/javascript/v8/javascriptinit.swg

```diff
@@ -69,8 +76,53 @@ void SWIGV8_INIT (SWIGV8_OBJECT exports_obj, SWIGV8_VALUE /*module*/, v8::Local<
 #if !defined(NODE_MODULE_VERSION) || NODE_MODULE_VERSION < 64
   v8::Local<v8::Context> context = SWIGV8_CURRENT_CONTEXT();
 #endif
+  v8::Isolate* isolate = SWIGV8_GETISOLATE(context);
+  SWIGV8_HANDLESCOPE();
+
+#if __cplusplus >= 201103L || (defined(_MSVC_LANG) && _MSVC_LANG >= 201103L)
+  static std::mutex guard;
+  const std::lock_guard<std::mutex> lock(guard);
+#endif
 
   SWIG_InitializeModule(context);
+
+#if (V8_MAJOR_VERSION-0) < 5
+  // Multiple threads are not supported, everything is saved in globals.
+  v8::Persistent<v8::FunctionTemplate>* v8_swig_types = NULL;
+  v8::Local<v8::Value> v8_swig_types_as_data;
```

Comment:
SWIG now only supports V8 5.0 and later, so this branch (and the conditional) aren't needed now (probably wasn't the case when you started this patch).